config/initializers/request_ip_address.rb

```diff
@@ -0,0 +1,31 @@
+Gemcutter::GEOIP_FIELDS = {
+  continent_code: "GEOIP_CONTINENT_CODE",
+  country_code: "GEOIP_COUNTRY_CODE",
+  country_code3: "GEOIP_COUNTRY_CODE3",
+  country_name: "GEOIP_COUNTRY_NAME",
+  region: "GEOIP_REGION",
+  city: "GEOIP_CITY"
+}.freeze
+
+Gemcutter::PROXY_TOKEN = ENV["RUBYGEMS_PROXY_TOKEN"].presence
+
+ActiveSupport.on_load(:action_dispatch_request) do
```

Comment:
Any details on this?

Reply:
This is used to lazily populate the geoip_info and user agent info in the `additional` json on events once they eventually get recorded